### ISSUE TITLE
Image() buffer docs are confusing

### ISSUE THREAD

**Issue by DavidWhaleMEF** [bbcmicrobit/micropython#437]
microbit.org-ticket: 1447

The following docs about using the Image with a buffer are confusing, and actually, I can't get them to work at all. Am I missing something??!!

http://microbit-micropython.readthedocs.io/en/latest/image.html?highlight=Image

First of all, it is not clear that kwargs are not supported in the docs, users are trying this form and getting an error:

![screen shot 2017-04-18 at 10 48 33](https://cloud.githubusercontent.com/assets/25079586/25124788/a17ee9f6-2424-11e7-9580-c8c396540730.png)

Secondly, the docs talk about passing in a buffer as an array of integers

![screen shot 2017-04-18 at 10 51 46](https://cloud.githubusercontent.com/assets/25079586/25124913/10234c80-2425-11e7-8728-a7c79f9037e2.png)

Python does not have an array type, it uses lists? Do you mean a list?

I tried with two forms of lists, both give a TypeError: object with buffer protocol required

![screen shot 2017-04-18 at 10 50 19](https://cloud.githubusercontent.com/assets/25079586/25124975/3a4b20e6-2425-11e7-8794-c4f667722649.png)

![screen shot 2017-04-18 at 10 51 23](https://cloud.githubusercontent.com/assets/25079586/25124984/3e1a2a1e-2425-11e7-9405-091774827f4b.png)

Can someone send me a working example of how this can be made to work? Thanks!

**Comment by deshipu** [comment]
Python does have an array type: https://docs.python.org/3.6/library/array.html

**Comment by dpgeorge** [comment]
> First of all, it is not clear that kwargs are not supported in the docs, users are trying this form and getting an error:

A lot of the built-in methods/functions in CPython don't allow keyword args, so we are not doing anything different here.  I guess it's just a case of make the docs explicit in this regard.  Maybe we should list the 4 differnt constructors for image, eg:
```python
Image() # create a blank 5x5 image
Image(s) # create an image by parsing the string (single char returns that glyph)
Image(w, h) # create a blank image of given size
Image(w, h, data) # create an image from the data
```

> Secondly, the docs talk about passing in a buffer as an array of integers
>  ...
> Can someone send me a working example of how this can be made to work?

You need to pass an object with the buffer protocol, something like a bytes object or bytearray will do it:
```python
Image(2, 2, b'\x00\x01\0x1\x00')
```
Note that the values you pass are raw bytes, so the above is like [[0,1],[1,0]].

**Comment by DavidWhaleMEF** [comment]
Thanks. Can you add this example to the docs, and also explain the lack of kwargs support, as both are actually not that clear even to me until you gave me a worked example!

**Comment by DavidWhaleMEF** [comment]
The docs say pass an array of integers - what you have above is an array of bytes?

**Comment by DavidWhaleMEF** [comment]
I see Python (and MicroPython) do have an array, thanks @deshipu - nothing comes up in the docs when I search for array though??

![screen shot 2017-04-18 at 14 07 21](https://cloud.githubusercontent.com/assets/25079586/25132161/6a0f7726-2440-11e7-94cf-aee2468046fd.png)

**Comment by DavidWhaleMEF** [comment]
I can confirm that this code works:

from microbit import *

i = Image(5,5,b'\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08')

display.show(i)

If you could update the docs with an example, and also explain the fact that you can't use kwargs, that will be great, thanks for your help.

**Comment by microbit-carlos** [comment]
Implemented in #582.
